### ISSUE TITLE
Reconcile/merge tasks from `issue/216-G`

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/git-subrepo#279]

**Comment by ingydotnet** [comment]
I found this command to be helpful:
```
git log --graph --date=short --left-right --pretty=format:'%h %ad %<(25) %an %s' origin/release/0.4.0...origin/issue/216-G
```

It show these commit exist in `origin/issue/216-G` but not in
`origin/release/0.4.0`:
```
> a356961 2017-05-17  Magnus Carlsson           Update dox
> 819be2e 2017-05-15  Magnus Carlsson           Add --method option to init/clone, add a 'config' command
> e37942c 2017-03-10  Magnus Carlsson           Detect multiple pulls, use -u flag to decide.
> c01cf67 2016-11-29  Ingy döt Net              Update the docs for pull and push
> 546ac15 2016-11-28  Ingy döt Net              Update docs to reflect how things should work
> 7cecf0b 2016-09-28  Magnus Carlsson           Update error messages when failing merge/rebase
> 8406a91 2016-09-28  Magnus Carlsson           First attempt to restore the rebase
> 3e77b1d 2016-09-24  Ingy döt Net              Fix env var bug in test/push.t
> 943c251 2016-08-19  Magnus Carlsson           Do not overwrite author information.
> 9cd4ce7 2016-07-05  Magnus Carlsson           Make it possible to specify commit messages.
> 131a106 2016-02-27  Magnus Carlsson           Redesign, trash the tree hash approach and use merges instead.
```

@grimmySwe, please go over these commits and add in the code that we want to be
in version 0.4.0.

You can push commits directly to `release/0.4.0` and I'll review them.

**Comment by grimmySwe** [comment]
@ingydotnet  I rebased in the relevant changes, so now release/0.4.0 will contain everything from issue/216-G. No merge commits :-)

I saw no reason to remove anything here. the `git subrepo config` command is questionable when we later add Merge/Rebase flag to pull but I find that it might be useful to allow users to set the method without running pull and --update. And it should not disturb anything else as it only operates on the .gitrepo file.
